emerging_optimizers/orthogonalized_optimizers/adaptive_orthogonalized_optimizer.py

```diff
@@ -0,0 +1,265 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Callable, Literal
+
+
+# TODO(@boxiangw): remove this once bump to python 3.12
+try:
+    from typing import override
+except ImportError:
+    from typing_extensions import override
+
+import torch
+from absl import logging
+from torch.optim.optimizer import ParamsT
+
+from emerging_optimizers import mixin as opt_mixin
+from emerging_optimizers import utils
+from emerging_optimizers.orthogonalized_optimizers import muon_utils
+from emerging_optimizers.orthogonalized_optimizers.muon import get_muon_scale_factor
+from emerging_optimizers.orthogonalized_optimizers.orthogonalized_optimizer import OrthogonalizedOptimizer
+
+
+_adaptive_args_doc = """params: Iterable of parameters to optimize or dicts defining parameter groups
+        lr: The learning rate used by the internal SGD.
+        momentum_beta: The momentum used by the internal SGD.
+        weight_decay: The weight decay used by the optimizer, default to be decoupled weight decay.
+            See Decoupled Weight Decay Regularization: https://arxiv.org/abs/1711.05101
+        use_nesterov: Whether to use Nesterov-style momentum in the internal SGD.
+        second_moment_method: Method to apply second moment. Options: "adamuon" (elementwise like AdamW), "normuon" (row/column-wise).
+        beta2: The exponential decay rate for second moment (like AdamW's beta2).
+        eps: Small constant for numerical stability in second moment normalization.
+        weight_decay_method: Method to apply weight decay, see :class:`~emerging_optimizers.mixin.WeightDecayMixin`
+            for more details.
+        fp32_matmul_prec: Precision of the matmul operations in optimizer states GEMM operations.
+"""
+
+
+class AdaptiveOrthogonalizedOptimizer(OrthogonalizedOptimizer):
+    """Orthogonalized optimizer with adaptive second moment (AdaMuon/NorMuon variants).
```

Comment:
It this meant to be a base class like OrthogonalizedOptimizer or a variant of Muon? Muon specific arguments don't belong to base class. Inherit from Muon if it meant to be a subclass of muon